zerver/lib/retention.py

```diff
@@ -228,18 +228,15 @@ def move_expired_direct_messages_to_archive(
     assert message_retention_days != -1
     check_date = timezone_now() - timedelta(days=message_retention_days)
 
-    recipient_types = (Recipient.PERSONAL, Recipient.DIRECT_MESSAGE_GROUP)
-
     # Archive expired direct Messages in the realm, including cross-realm messages.
-    # Uses index: zerver_message_realm_recipient_date_sent
+    # Uses index: zerver_message_realm_date_sent
     query = SQL(
         """
     INSERT INTO zerver_archivedmessage ({dst_fields}, archive_transaction_id)
         SELECT {src_fields}, {archive_transaction_id}
         FROM zerver_message
-        INNER JOIN zerver_recipient ON zerver_recipient.id = zerver_message.recipient_id
         WHERE zerver_message.realm_id = {realm_id}
-            AND zerver_recipient.type in {recipient_types}
+            AND NOT zerver_message.is_channel_message
             AND zerver_message.date_sent < {check_date}
```

Comment:
we could also consider adding some index that'd help this query further? e.g. a partial index on `(realm_id, date_sent` where `is_channel_message=false`? .